src/subject/subject.controller.ts

```diff
@@ -19,10 +37,33 @@ export class SubjectController {
   ): Promise<IResponsePaginate> {
     return await this.subjectService.findAll(query);
   }
+
   @UseGuards(JwtAuthGuard)
   @ApiBearerAuth()
   @Get(':id')
   async findOne(@Param('id') id: number) {
     return await this.subjectService.listSubjects(id);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @ApiBearerAuth()
+  @Roles(Role.Coordinator)
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ApiOperation({
+    summary: 'Atualiza o status de uma responsabilidade para 3 (finalizada)',
+  })
+  @Patch('responsability/:id/end')
+  async endResponsability(@Param('id') id: number) {
+    try {
+      return await this.endResponsabilityCommand.execute(id);
+    } catch (error) {
+      if (error instanceof ResponsabilityNotFoundException) {
+        throw new NotFoundException(error.message);
+      } 
+
+      if (error instanceof AlreadyFinishedException) {
+        throw new PreconditionFailedException(error.message);
+      }
```

Comment:
no final do catch você sempre tem que colocar o `throw error` porque se o erro for desconhecido nós precisamos lança-lo também.

Reply:
@nabsonp pode marcar cmo resolved aqui? pq modifiquei a linha embaixo, acho q o github n pegou automatico